assets/training/finetune_acft_image/components/pipeline_components/instance_segmentation/spec.yaml

```diff
@@ -0,0 +1,413 @@
+    type: number
+    optional: true
+
+  step_lr_step_size:
+    description: Value of step size when learning rate scheduler is step. Please check for https://learn.microsoft.com/azure/machine-learning/reference-automl-images-hyperparameters more information.
+    type: integer
+    optional: true
+
+  tile_grid_size:
+    description: The grid size to use for tiling each image. Should be passed as a string in '3x2' format. Example --tile_grid_size '3x2'
+    type: string
+    optional: true
+
+  tile_overlap_ratio:
+    description: Overlap ratio between adjacent tiles in each dimension. Must be float in the range of [0, 1).
+    type: number
+    optional: true
+    min: 0
+    max: 1
+
+  tile_predictions_nms_threshold:
+    description: The IOU threshold to use to perform NMS while merging predictions from tiles and image. Used in validation/ inference. Must be float in the range of [0, 1].
+    type: number
+    optional: true
+    min: 0
+    max: 1
+
+  training_batch_size:
+    description: Training batch size.
+    type: integer
+    optional: true
+    min: 1
+
+  validation_batch_size:
+    description: Validation batch size.
+    type: integer
+    optional: true
+    min: 1
+
+  validation_iou_threshold:
+    description: IOU threshold for box matching when computing validation metrics. Must be a float in the range [0.1, 1].
+    type: number
+    optional: true
+    min: 0.1
+    max: 1
+
+  validation_metric_type:
+    description: Metric computation method to use for validation metrics. Must be none, coco, voc, or coco_voc.
+    type: string
+    optional: true
+    default: 'voc'
+    enum: ['none', 'coco', 'voc', 'coco_voc']
+
+  warmup_cosine_lr_cycles:
+    description: Value of cosine cycle when learning rate scheduler is warmup_cosine. Please check for https://learn.microsoft.com/azure/machine-learning/reference-automl-images-hyperparameters more information.
+    type: number
+    optional: true
+
+  warmup_cosine_lr_warmup_epochs:
+    description: Value of warmup epochs when learning rate scheduler is warmup_cosine. Please check for https://learn.microsoft.com/azure/machine-learning/reference-automl-images-hyperparameters more information.
+    type: integer
+    optional: true
+
+  weight_decay:
+    description: Value of weight decay used by the optimizer.
+    type: number
+    optional: true
+    min: 0
+    max: 1
+
+outputs:
+  pytorch_model_folder:
+    type: custom_model
+    description: Trained pytorch model.
+  mlflow_model_folder:
+    type: mlflow_model
+    description: The trained MLFlow model.
+
+jobs:
+  framework_selector:
+    type: command
+    component: azureml:image_framework_selector:0.0.6
+    compute: ${{parent.inputs.compute_model_import}}
+    inputs:
+      task_type: ${{parent.inputs.task_type}}
+      model_name: ${{parent.inputs.model_name}}
+
+  image_instance_segmentation_runtime_component:
+    type: command
+    component: azureml:train_instance_segmentation_model:0.0.4
```

Comment:
Can we push only the runtime components from previous release to prod, without the rest of components ? Otherwise, should this version change ?

Reply:
Yes, we should be able to. 
We have never pushed these to prod.